tests/platform_tests/test_cont_warm_reboot.py

```diff
@@ -0,0 +1,125 @@
+
+
+def check_reboot_type(dut, reboot_type=None):
+    """
+    Perform a match of reboot-cause and reboot-trigger
+    @param dut: The AnsibleHost object of DUT.
+    """
+    if reboot_type is not None:
+        logging.info("Check reboot cause")
+        assert wait_until(MAX_WAIT_TIME_FOR_REBOOT_CAUSE, 20, check_reboot_cause, dut, reboot_type), \
+            "got reboot-cause failed after rebooted by %s" % reboot_type
+
+        if reboot_ctrl_dict[reboot_type]["test_reboot_cause_only"]:
+            logging.info("Further checking skipped for %s test which intends to verify reboot-cause only" % reboot_type)
+            return
+
+
+def check_interfaces_and_transceivers(dut, interfaces):
+    """
+    Perform a check of transceivers, LAGs and interfaces status
+    @param dut: The AnsibleHost object of DUT.
+    @param interfaces: DUT's interfaces defined by minigraph
+    """
+    logging.info("Wait %d seconds for all the transceivers to be detected" % MAX_WAIT_TIME_FOR_INTERFACES)
+    assert wait_until(MAX_WAIT_TIME_FOR_INTERFACES, 20, check_interface_information, dut, interfaces), \
+        "Not all transceivers are detected or interfaces are up in %d seconds" % MAX_WAIT_TIME_FOR_INTERFACES
+
+    logging.info("Check transceiver status")
+    check_transceiver_basic(dut, interfaces)
+
+    logging.info("Check LAGs and interfaces status")
+    checks.check_interfaces(dut)
+
+
+def check_neighbors(dut):
+    """
+    Perform a BGP neighborship check.
+    @param dut: The AnsibleHost object of DUT.
+    """
+    logging.info("Check BGP neighbors status. Expected state - established")
+    bgp_facts = dut.bgp_facts()['ansible_facts']
+    mg_facts  = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
+
+    for value in bgp_facts['bgp_neighbors'].values():
+        # Verify bgp sessions are established
+        assert value['state'] == 'established'
+        # Verify locat ASNs in bgp sessions
+        assert value['local AS'] == mg_facts['minigraph_bgp_asn']
+
+    for v in mg_facts['minigraph_bgp']:
+        # Compare the bgp neighbors name with minigraph bgp neigbhors name
+        assert v['name'] == bgp_facts['bgp_neighbors'][v['addr'].lower()]['description']
+        # Compare the bgp neighbors ASN with minigraph
+        assert v['asn'] == bgp_facts['bgp_neighbors'][v['addr'].lower()]['remote AS']
+
+
+def test_cont_warm_reboot(duthost, localhost, conn_graph_facts, cont_reboot_limit):
+    """
+    @summary: This test case is to perform continuous warm reboot in a row
+    """
+    asic_type = duthost.facts["asic_type"]
+    if asic_type in ["mellanox"]:
+        issu_capability = duthost.command("show platform mlnx issu")["stdout"]
+        if "disabled" in issu_capability:
+            pytest.skip("ISSU is not supported on this DUT, skip this test case")
+
+    for _ in range(cont_reboot_limit):
+        reboot_and_check(localhost, duthost, conn_graph_facts["device_conn"], reboot_type=REBOOT_TYPE_WARM)
```

Comment:
You should add another parameter to control a delay after each reboot.

Reply:
Sure, I will add it. Do you have a suggestion for the default value for this delay parameter?

Reply:
Added a new delay param with default wait between reboots set to 10s.

Reply:
10 seconds is too short. Default should be 5 minutes.